src/main/java/nl/YEA/view/MeerkeuzeVraagInvulScherm.java

```diff
@@ -8,6 +14,82 @@ public MeerkeuzeVraagInvulScherm(int vraag) {
 
     @Override
     public void show() {
-        InOutputUtil.getInstance().printNl("meerkeuze");
+        InOutputUtil io = InOutputUtil.getInstance();
+        Singleton singleton = Singleton.getInstance();
+        MeerkeuzeVraagController meerkeuzeVraagController = singleton.getMeerkeuzeVraagController();
+        String vraagbeschrijving = meerkeuzeVraagController.getVraagBeschrijving(vraagNr);
+        List<String> gekozenAntwoorden = new ArrayList<>();
+        List<String> mogenlijkeAntwoorden = meerkeuzeVraagController.getMogenlijkeAntwoorden(vraagNr);
+        int max = meerkeuzeVraagController.getMaxKeuzes(vraagNr);
+        int min = meerkeuzeVraagController.getMinKeuzes(vraagNr);
+        boolean running = true;
+        while (running){
+            io.printNl(vraagbeschrijving);
+            if (gekozenAntwoorden.size()>0) {
+                io.print("reeds gekozen antwoorden:\n");
+                for (String keuze : gekozenAntwoorden){
+                    io.print(keuze+"\n");
+                }
+            }
+            io.print("min: "+min+"  max: "+max+"\n");
+            io.print("maak u keuze:\n");
+            List<String> hedenMogenlijkeAntworden = new ArrayList<>(mogenlijkeAntwoorden);
+            hedenMogenlijkeAntworden.removeAll(gekozenAntwoorden);
+            printAntwoordKeuzes(io, hedenMogenlijkeAntworden);
+            int i = hedenMogenlijkeAntworden.size()+1;
+            if (gekozenAntwoorden.size()>0) {
+                io.print("[" + i + "] : wis antwoorden\n");
+                i++;
+            }
+            if (gekozenAntwoorden.size()>=min){
+                io.print("[" + i + "] : akkoord\n");
+            }
+            running = listenforKeuze(io,hedenMogenlijkeAntworden, gekozenAntwoorden, min,max);
+        }
+        int[] antwoordIndexes = new int[gekozenAntwoorden.size()];
+        for (int i = 0 ;i<antwoordIndexes.length;i++){
+            antwoordIndexes[i]= mogenlijkeAntwoorden.indexOf(gekozenAntwoorden.get(i));
+        }
+        singleton.getMeerkeuzeAntwoordController().addToList(vraagNr,antwoordIndexes);
+
+    }
+
+    private List<String> printAntwoordKeuzes(InOutputUtil io, List<String> hedenMogenlijkeAntworden){
+        int i = 1;
+        for (String keuze:hedenMogenlijkeAntworden){
+            io.print("["+i+"] : "+keuze+"\n");
+            i++;
+        }
+        return hedenMogenlijkeAntworden;
+    }
+
+    private boolean listenforKeuze(InOutputUtil io, List<String> hedenMogenlijkeKeuzes, List<String> alGekozen,
+                                   int min, int max){
+        String input = io.getNextLine();
+        int keuze = 0;
+        try {
+            keuze = Integer.parseInt(input);
+        }catch (NumberFormatException e){
+            io.print("geen nummer probeer opniew:\n");
+            return true;
+        }
```

Comment:
Misschien netter om bovenstaand stuk (regel 68 t/m 75) in losse methode te zetten?